### PR TITLE
Ats Analytics Adapter:  handle preflight request error & increase _lr_sample_rate cookie expiration

### PR DESCRIPTION
**Pull request by mamatic**
## Type of change

- [x] Bugfix
- [x] Improvement 

## Description of change
- Increase expiration time for _lr_sampling_rate cookie to 24h.
- Handle error for preflight request, if we get error set _lr_sampling_rate cookie value to 0.

- test parameters for validating bids

## Other information
@mkendall07 , @ChrisHuie

### CODE REVIEW

**Comment by mamatic** [comment]
@ChrisHuie I added unit tests

**Comment by mamatic** [comment]
Hi @osazos, can you please review pull request ? Thanks

**Review by osazos — COMMENTED** [review]
Your code is ok, but can you increase your coverage rate to at least 80%?

File: modules/atsAnalyticsAdapter.js [hunk 1]
```diff
@@ -276,16 +267,23 @@ function sendDataToAnalytic () {
...
-    if (data && samplingRate && atsAnalyticsAdapter.shouldFireRequest(samplingRateNumber) && envelopeSourceCookieValue != null) {
-      sendDataToAnalytic();
-    }
-  }, undefined, { method: 'GET', crossOrigin: true });
+  ajax(preflightUrl + atsAnalyticsAdapter.context.pid,
+    {
+      success: function (data) {
+        let samplingRateObject = JSON.parse(data);
+        logInfo('ATS Analytics - Sampling Rate: ', samplingRateObject);
+        let samplingRate = samplingRateObject['samplingRate'];
```
[the remark this answers is not in the document]
Comment (mamatic) [reply modules/atsAnalyticsAdapter.js]: > Hi @mamatic,
> 
> Your code is ok, but can you increase your coverage rate to at least 80%?

Sure, I added few more tests, coverage is increased. And I changed const and dot notation. Thanks

File: modules/atsAnalyticsAdapter.js [hunk 2]
```diff
@@ -356,6 +354,13 @@ atsAnalyticsAdapter.shouldFireRequest = function (samplingRate) {
 atsAnalyticsAdapter.getUserAgent = function () {
   return window.navigator.userAgent;
 };
+
+atsAnalyticsAdapter.setSamplingCookie = function (samplRate) {
+  let now = new Date();
```
[the remark this answers is not in the document]
Comment (mamatic) [reply modules/atsAnalyticsAdapter.js]: Done

**Review by osazos — APPROVED** [review]
LGTM

**Pull request merged**